docs: replace dataset discovery page with NeuroBlueprint data input guide (#90)
neuroinformatics-unit/photon-mosaic: docs/source/user_guide/configuration.md

### User Configuration
On first run, photon-mosaic will create a user config at `~/.photon_mosaic/config.yaml` if it does not exist. This serves as your default configuration.

> The pipeline operates on a single project directory that follows the [NeuroBlueprint](https://neuroblueprint.neuroinformatics.dev/) specification. Raw data lives under `<project_path>/rawdata/`, and derivatives are written next to it under `<project_path>/derivatives/`. See the [data input documentation](data_input.md) for the required directory layout.

Reviewer: ```suggestion
`photon-mosaic` expects you to organise your data in a project directory that follows the [NeuroBlueprint](https://neuroblueprint.neuroinformatics.dev/) specification. Raw data lives under `<project_path>/rawdata/`, and derivatives are written next to it under `<project_path>/derivatives/`. See the [data input documentation](data_input.md) for the required directory layout.
```

## Key Configuration Notes

> ### Data input

Reviewer: Should we remove this section entirely as it pretty much repeats a sentence from before in its own section?

Author: Yes I agree
